### ISSUE TITLE
always get output nan

### ISSUE THREAD

**Issue by jesse1222** [overfitover/pixor_pytorch#6]
When I choose much larger batchsize, I will get nan immediately.  But if I choose batchsize=1, it will work at the beginning.  However, it will become nan eventually.  The training process is all the same as the default.

**Comment by overfitover** [comment]
@jesse1222 
可能你的数据集中某些数据有些问题。你可以尝试用小数据集测试一些。另外把学习率改小些试试。出现nan有很多可能，我测试的时候没有出现这种情况，你可以多尝试一下。还有检查自己的环境是不是和要求的一样。

**Comment by overfitover** [comment]
补充一下，你可以修改一下损失函数，用smooth L1 函数试试。这个函数可以预防梯度爆炸。在loss里面修改

**Comment by jesse1222** [comment]
谢谢你的回覆
我是使用KITTI的3D数据集，使用的frame是能看完全部数据的大小，还是这部分会有问题？
另外code原先就是l1_smooth了
目前我是先将loc_loss+=F.....那边直接先除以batchsize*image_size，希望减少overflow的可能

**Comment by overfitover** [comment]
出现nan之前loss是在逐渐变大还是突然nan了？你需要定位一下。我看了一下代码，loss那块应该没问题，你先做个小数据集测试一下，可能你用的数据集有些损坏。你要把具体的情况贴出来看看。另外还有个分支的，你也可以试试另一个分支的情况。

**Comment by jesse1222** [comment]
我发现到当我把batchsize比较大的时候，cls_loss会容易变为inf，但这是不是数据造成的问题我会再去看看

**Comment by zuowanghanguang** [comment]
如果你是只有某个数据训练时出现nan，才说明是你的数据的问题：框超出的区域或者数据中有脏数据。你是cls出现nan,有一个可能的数据问题是该数据是没有任何标签的，所以logic函数出现了0。但是如果你是从该步开始之后全是nan，那就是你设置和网络的问题了

**Comment by jesse1222** [comment]
@zuowanghanguang 
的确很有可能是你说的问题，但是应该是因为logic函数出现0，所以log导致出现inf，所以接下来的iteration都会被这个inf的loss，而变成nan

**Comment by overfitover** [comment]
@jesse1222 
你可以修改focal loss 函数或是改用交叉熵函数试试。修改后的函数还会出现nan的情况吗？

**Comment by andrearama** [comment]
I also experienced this issue. As mentioned, putting batch size of one helps. 
Furthermore, sometime there was the case in which the focal loss (defined in Loss/loss.py) gives inf.
This is because x_t is allowed to become 0, and the consequent log would bring it to inf.
For me was enough to add a small quantity to it to avoid such problem:
`x_t = x * (2 * y - 1) + (1 - y) +0.00000000001`
